source/gloperate/source/pipeline/Stage.cpp

```diff
@@ -370,6 +388,21 @@ AbstractSlot * Stage::createSlot(const std::string & slotType, const std::string
     return nullptr;
 }
 
+std::string Stage::qualifiedName() const
+{
+    std::string path = name();
+
+    Pipeline * pipeline = this->parentPipeline();
+    while (pipeline)
```

Comment:
Since name and parent are defined in `AbstractProperty`, i guess that would be the place.